rotkehlchen/utils/mixins/penalizable_oracle.py

```diff
@@ -0,0 +1,36 @@
+import dataclasses
+
+from rotkehlchen.types import Timestamp
+from rotkehlchen.utils.misc import ts_now
+
+ORACLE_PENALTY_THRESHOLD_COUNT = 5
+ORACLE_PENALTY_TS = 1800
+
+
+@dataclasses.dataclass(init=True, repr=True, eq=False, order=False, unsafe_hash=False, frozen=False)  # noqa: E501
+class PenaltyInfo:
+    last_penalized_ts: Timestamp
+    query_failures_count: int
+
+    def note_failure_or_penalize(self) -> None:
+        """
+        This function determines whether an oracle should be punished or not.
+
+        It is called when an oracle fails, if the failure count reaches a threshold,
+        it is punished for a duration i.e. skipped. Otherwise, the failure count is incremented.
+        """
+        if self.query_failures_count >= ORACLE_PENALTY_THRESHOLD_COUNT:
+            self.last_penalized_ts = ts_now()
+            self.query_failures_count = 0
+        else:
+            self.query_failures_count += 1
+
+
+class PenalizablePriceOracleMixin:
+    """This class represents oracle that can be penalized due to failures."""
+    def __init__(self) -> None:
```

Comment:
Add a todo. To Make `ORACLE_PENALTY_THRESHOLD_COUNT` and `ORACLE_PENALTY_TS` as configurable and as parameters here.

This way we can allow the user to edit them. Make an issue about this and reference it here.